smolyay/samples.py

```diff
@@ -0,0 +1,362 @@
+import abc
+import numpy
+
+
+class UnidimensionalPointSet(abc.ABC):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points to be used as the sampling points
+    in a tensor product grid or sparse grid. These unidimensional points are
+    associated with certain polynomial families, and the combination of
+    unique points and polynomial families are the basis for different types
+    of quadrature rules that are used in numerical integration and in
+    approximation.
+    The natural domain of the points specified by :param:``domain`` is
+    the domain in which all the possible points in the set lies within.
+    ``points`` is the list of points that the UnidimensionalPointSet
+    is expected to store.
+    ``domain`` is the domain of the points generated by the point set.
+    """
+
+    def __init__(self):
+        self._points = None
+        self._valid_cache = True
+
+    @property
+    @abc.abstractmethod
+    def domain(self):
+        """list: Domain the sample points come from."""
```

Comment:
Let's make this a numpy array like we did for the basis functions
```suggestion
        """numpy.ndarray: Domain of the `points`."""
```